### PR TITLE
build: add release-it to facilitate builds

### LINKED ISSUE #267: Deprecate this repo in favour of an active fork
The organisation I work for has moved from SAML to OIDC for authentication, and hence I have less motivation to be active on this repo. Since @yaronn is still the admin and owner of this repo, all requests to be given write rights for the repo must go through him, even though he has been inactive on xml-crypto for a while.

This package is still in demand though, as evidenced by recent proposals to add a new KeyInfo implementation, and by its use in node-saml, of which @cjbarth is an active maintainer. We could ask for a transfer of this repo to a new organisation, but @yaronn is inactive and we don't know when he would be able to accede to the request.

Given this, I would like to propose that:
- a new repo based on this one is created and actively managed by current users of this package; 
- the xml-crypto npm package is published from that new repo, and;
- a prominent notice is added to the README of this repo indicating in no uncertain terms that this repo is inactive and to use the new repo.

@cjbarth, what do you think?

### LINKED ISSUE #274: Make releases with `release-it`
It would be nice if one of the first orders of business for `xml-crypto` would be to get it working with `release-it` to make releasing a 1 minutes process as it currently is with `node-saml` and `passport-saml`.

_Originally posted by @cjbarth in https://github.com/node-saml/xml-crypto/issues/267#issuecomment-1385663484_

### PR DESCRIPTION
**Pull request by LoneRifle**
## Problem

Releases are currently manual, tedious and prone to human error. While fully automated releases would be great, creating and maintaining such scaffolding would require planning. A much easier alternative would be to automate as much of the process for the publisher as possible

Closes #274 

## Solution

Port release scaffolding from passport-saml for consistent publishing across all repos in the node-saml organization

- `npm install --save-dev release-it gren`
- add release-it config and changelog run script from passport-saml

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,9 @@
   },
   "devDependencies": {
     "ejs": "3.1.7",
-    "nodeunit": "^0.11.3"
+    "gren": "^0.0.1",
```
Comment (cjbarth) [inline package.json:18]: I think if you look at the `package.json` from `node-saml`, you'll see that a different package is used: https://github.com/node-saml/node-saml/blob/f86252bcbaa4435159d121d2e342ec89f94ad183/package.json#L68

**Comment by cjbarth** [comment]
Are we going to release this on NPM using the @node-saml organization too, or will we continue with the package as it currently is on NPM? What we did with `node-saml` and `passport-saml` is we put up a deprecation notice for the unscoped packages and re-published under an NPM organization. It might make sense to do that here.

File: package.json (hunk 1 above)
Comment (LoneRifle) [reply package.json]: whoops! fixed in amended commit

**Comment by LoneRifle** [comment]
@cjbarth It's up to you! We've just made one major release with breaking changes not too long ago, so we may as well rename the package (though we would have to copy over the DT types accordingly), since users will have to test before upgrading anyway.

**Comment by djaqua** [comment]
I'm on board with making releases easier. Also, do we have someplace where we can manage release goals or have a discussion about the direction of this project?

**Comment by LoneRifle** [comment]
@cjbarth @djaqua - I would like to have this PR merged so that we can move ahead with releases. As suggested, I will rename the package to `@node-saml/xml-crypto`

**Review by cjbarth — APPROVED** [review]

**Comment by cjbarth** [comment]
@LoneRifle , I made a few changes to this to use the latest package for generating changelogs and then added the first changelog. Have a look and LMK what you think. If you approve, I'll land it.

**Comment by LoneRifle** [comment]
Can't approve something I raise myself! Go ahead and merge.

**Pull request merged**